lib/perl/Genome/Model/Tools/Speedseq/Base.pm

```diff
@@ -0,0 +1,290 @@
+package Genome::Model::Tools::Speedseq::Base;
+
+use strict;
+use warnings;
+
+use Genome;
+use Carp qw(confess);
+use Data::Dumper;
+
+class Genome::Model::Tools::Speedseq::Base {
+    is => 'Command::V2',
+    is_abstract => 1,
+    attributes_have => [
+        tool_param_name => {
+            is => 'String',
+            is_optional => 1,
+        },
+        tool_bare_arg_position => {
+            is => 'Integer',
+            is_optional => 1,
+        },
+        tool_input_file => {
+            is => 'Boolean',
+            is_optional => 1,
+        },
+    ],
+    has_param => [
+        version => {
+            is => 'Version',
+            doc => 'SpeedSeq version to be used',
+            is_optional => 1,
+        },
+        config_file => {
+            is => 'Text',
+            doc => 'path to speedseq.config file (default: same directory as speedseq)',
+            is_optional => 1,
+            tool_param_name => 'K',
+            tool_input_file => 1,
+        },
+        verbose => {
+            is => 'Boolean',
+            doc => 'verbose',
+            is_optional => 1,
+            tool_param_name => 'v',
+        },
+    ],
+    has_optional_output => [
+        output_files => {
+            is_many => 1,
+        },
+        input_files => {
+            is_many => 1,
+        },
+    ],
+};
+
+sub help_detail {
+    "A flexible framework for rapid genome analysis and interpretation"
+}
+
+my %TOOL_VERSIONS = (
+    'test'     => '/gscmnt/gc2719/halllab/bin/speedseq',
```

Comment:
This version fails on very small input files (e.g. for testing). I have worked with Colby on getting it fixed and he has an updated version at ` /gscmnt/gc2719/halllab/users/cchiang/bin/speedseq`. He will also update  the official version at the above path in a week or so once their current alignments have finished. We should make sure that we package a speedseq version that includes commit `6645b100a4d6dc329409d9fe74c41e0269443a4f`.